pkg/test/flags.go

```diff
@@ -17,23 +17,86 @@ package test
 import (
 	"flag"
 	"fmt"
+	"os"
+	"strings"
 
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+	"istio.io/istio/pkg/log"
 	"istio.io/istio/pkg/test/impl/driver"
 )
 
 var arguments = driver.DefaultArgs()
+var showHelp = false
+var logOptions = log.DefaultOptions()
+
+// The command we use to parse flags.
+var cmdForLog = &cobra.Command{}
 
 // init registers the command-line flags that we can exposed for "go test".
 func init() {
-	// TODO: Register logging flags.
+	// First, attach the logOptions to the cobra command. This just establishes their relationship.
+	logOptions.AttachCobraFlags(cmdForLog)
+
+	// Attach our own flags to Cobra flags. This is how we will read them.
+	attachFlags(cmdForLog.PersistentFlags().StringVar, cmdForLog.PersistentFlags().BoolVar)
+
+	// Apply the flags captured in cobra to the regular Go flags, so that Go flag code won't barf when
+	// it sees these flags.
+	cmdForLog.PersistentFlags().VisitAll(func(f *pflag.Flag) {
```

Comment:
I haven't seen this before.  Would the alternative be to just define them as flags above instead? Is there any advantage to doing it this way?

Reply:
This also captures the logging flags, which got set by logOptions.AttachCobraFlags().

The problem is, we use both flag parsing models (Go's own flags, and cobra), mainly because logging framework uses Cobra, and go tests uses Go's own. This allows Go's parser to accept the flags defined on the cobra side.